src/backend/distributed/ddl/foreign_constraint.c

```diff
@@ -0,0 +1,369 @@
+/*-------------------------------------------------------------------------
+ *
+ * foreign_constraint.c
+ *
+ * This file contains functions to perform useful operations on foreign
+ * keys.
+ *
+ * Copyright (c) 2014-2016, Citus Data, Inc.
```

Comment:
Two minor notes:

1. We should probably use 2018.

2. `useful operations` seems too broad. Maybe: `This file contains functions to create, alter and drop foreign constraints on distributed tables.`